client/data/site-verticals/use-site-verticals-query.ts

```diff
@@ -0,0 +1,50 @@
+import { useQuery, UseQueryResult, QueryKey } from 'react-query';
+import wpcom from 'calypso/lib/wp';
+import { mapManyVerticalsApiToVertical } from './utils';
+import type {
+	SiteVerticalsVerticalApi,
+	SiteVerticalsVertical,
+	SiteVerticalsQueryParams,
+	SiteVerticalsQueryOptions,
+} from './types';
+
+const defaults = {
+	term: '',
+	limit: 10,
+	skip_synonyms: false,
+};
+
+const useSiteVerticalsQuery = (
+	fetchOptions: SiteVerticalsQueryParams = {},
+	queryOptions: SiteVerticalsQueryOptions = {}
+): UseQueryResult< SiteVerticalsVertical[] > => {
+	return useQuery(
+		getCacheKey( fetchOptions.term || '' ),
+		() => fetchSiteVerticals( { ...defaults, ...fetchOptions } ),
+		{
+			select: mapManyVerticalsApiToVertical,
```

Comment:
I think this should come from the query options to let the developer decide the final format. But I'm not sure whether it would affect the cache or not.

Reply:
I've decoupled it and move the mapping/normalization to the UI component.